### PR TITLE
Add the possibility to use htc-vive in OculusModule

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

File: modules/Oculus_module/src/OculusModule.cpp [hunk 1]
```diff
@@ -932,6 +983,42 @@ bool OculusModule::updateModule()
...
+                    yError() << "[OculusModule::updateModule] I will not start the walking. Please "
+                                "try to start again.";
+                    return true;
+                }
+
+                // get only the yaw axis
+                iDynTree::Rotation tempRot(getRotation(oculusInertial_T_oculusRoot));
+                double yaw = 0;
+                double dummy = 0;
+                tempRot.getRPY(dummy, dummy, yaw);
```
Comment (S-Dafarra) [inline modules/Oculus_module/src/OculusModule.cpp:1012]: Note that here you are assuming that the Z axis is parallel to gravity, but according to https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#reference-spaces it is the Y axis pointing upward

File: modules/Oculus_module/src/OculusModule.cpp [hunk 2]
```diff
@@ -23,6 +23,29 @@
...
 #include <functional>
 
+Eigen::Ref<Eigen::Matrix<double, 3, 3, Eigen::RowMajor>>
+getRotation(yarp::sig::Matrix& m)
+{
+    assert(m.rows() == 4 && m.cols() = 4 && "getRotation expects a 4x4 matrix");
+    return Eigen::Map<Eigen::Matrix<double, 4, 4, Eigen::RowMajor>>(m.data()).topLeftCorner<3,3>();
+}
+
+auto getPosition(yarp::sig::Matrix& m)
```
Comment (S-Dafarra) [inline modules/Oculus_module/src/OculusModule.cpp:33]: I am not sure if this is ``c++14``.

File: modules/Oculus_module/include/OculusModule.hpp [hunk 3]
```diff
@@ -79,6 +79,7 @@ class OculusModule : public yarp::os::RFModule
     bool m_useVirtualizer; /**< True if the virtualizer is used in the retargeting */
     bool m_useXsens; /**< True if the Xsens is used in the retargeting */
     bool m_useIFeel; /**< True if the iFeel is used in the retargeting */
+    bool m_useVive; /**< True if the iFeel is used in the retargeting */
```
Comment (S-Dafarra) [inline modules/Oculus_module/include/OculusModule.hpp:82]: Copy paste

File: modules/Oculus_module/src/OculusModule.cpp [hunk 4]
```diff
@@ -598,8 +636,21 @@ bool OculusModule::getTransforms()
...
+                iDynTree::toEigen(m_oculusRoot_T_headOculus)
+                    = iDynTree::toEigen(m_oculusRoot_T_oculusInertial)
+                      * iDynTree::toEigen(m_oculusRoot_T_headOculus);
+            }
         }
     }
+
+    // m_oculusHeadsetPoseInertial is a 6d std::vector here I'm getting the first three elements
+    Eigen::Map<Eigen::Vector3d>(m_oculusHeadsetPoseInertial.data())
+        = getPosition(m_oculusRoot_T_headOculus);
```
Comment (S-Dafarra) [inline modules/Oculus_module/src/OculusModule.cpp:652]: What about the rotation?

File: modules/Oculus_module/src/OculusModule.cpp (hunk 1 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp:1012]: `oculusInertial_T_oculusRoot` arrives from the transform server so it should contain a rotation matrix then we get only the yaw (idyntree definition). I think this does not depends on the name of the frame of openxr.

File: modules/Oculus_module/src/OculusModule.cpp (hunk 4 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp:655]: Ops. Here you have 
f96166e

File: modules/Oculus_module/include/OculusModule.hpp (hunk 3 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/include/OculusModule.hpp]: 7ab3ab4

File: modules/Oculus_module/src/OculusModule.cpp (hunk 2 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp:33]: 😭  I'm not able to make the code working without auto. I should return the type of 
```
Eigen::Map<Eigen::Matrix<double, 4, 4, Eigen::RowMajor>>(m.data()).topRightCorner<3,1>()
```

**Comment by S-Dafarra** [comment]
Closing this in favor of https://github.com/robotology/walking-teleoperation/pull/62

**Pull request closed without merging**